mypy.ini

```diff
@@ -0,0 +1,104 @@
+[mypy]
+show_error_codes = True
+warn_unused_configs = True
+warn_redundant_casts = True
+plugins = pydantic.mypy
+
+[mypy-matplotlib.*]
+ignore_missing_imports = True
+
+[mypy-mpl_toolkits.*]
+ignore_missing_imports = True
+
+[mypy-astropy.*]
+ignore_missing_imports = True
+
+[mypy-numpy.*]
+ignore_missing_imports = True
+
+[mypy-scipy.*]
+ignore_missing_imports = True
+
+[mypy-pyarrow.*]
+ignore_missing_imports = True
+
+[mypy-pandas.*]
+ignore_missing_imports = True
+
+[mypy-sqlalchemy.*]
+ignore_missing_imports = True
+
+[mypy-erfa.*]
+ignore_missing_imports = True
+
+[mypy-yaml.*]
+ignore_missing_imports = True
+
+[mypy-deprecated.*]
+ignore_missing_imports = True
+
+[mypy-pkg_resources.*]
+ignore_missing_imports = True
+
+[mypy-sklearn.*]
+ignore_missing_imports = True
+
+# Don't check LSST packages generally or even try to import them, since most
+# don't have type annotations.
+
+[mypy-lsst.*]
+ignore_missing_imports = True
+ignore_errors = True
+
+# Do expect to import type annotations from LSST middleware dependencies that
+# should have them, but don't re-check them as they have their own mypy.ini
+# files with their own exceptions.
+
+[mypy-lsst.utils.*]
+ignore_missing_imports = False
+ignore_errors = True
+
+[mypy-lsst.resources.*]
+ignore_missing_imports = False
+ignore_errors = True
+
+# Check all of daf.butler...
+
+[mypy-lsst.daf.butler.*]
+ignore_missing_imports = False
+ignore_errors = False
+disallow_untyped_defs = True
+disallow_incomplete_defs = True
+strict_equality = True
+warn_unreachable = True
+warn_unused_ignores = True
+
+# ...except the modules and subpackages below (can't find a way to do line
+# breaks in the lists of modules).
+
+[mypy-lsst.daf.butler.core.config]
```

Comment:
You should remove these butler ignores I think. You should not be trying to deal with butler hierarchies.

Reply:
Are you referring to lines 79 - 98, or the preceeding lines 67 - 98 as well?

Reply:
79-98 are specifically for daf_butler internals and this is not daf butler. The ones above really depends on what packages you are trying to use that you want mypy to ignore or not. You will likely want to mask out all the packages you use that don't themselves use mypy but since we don't know what packages you are using yet you don't really know what to put in this file

Reply:
Ok, thanks. I've removed lines 79-98, and will reassess the block above following an initial large code merge into the repo.